include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -49,8 +49,9 @@ __pattern_walk1(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIte
         oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read_write, _ForwardIterator>();
     auto __buf = __keep(__first, __last);
 
-    oneapi::dpl::__par_backend_hetero::__parallel_for(__exec, unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f},
-                                                      __n, __buf.all_view())
+    oneapi::dpl::__par_backend_hetero::__parallel_for(::std::forward<_ExecutionPolicy>(__exec),
```

Comment:
Could you please do the same for https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h#L234 and https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h#L591-L592 as well?

Reply:
I agree with you comment for numeric_impl, but disagree for radix_sort. Semantically if we use policy with std::forward on the caller side you don't know what callee does with it. Thus, we cannot say what state policy has after forwarding since it may cause move inside. if we want to reuse policy further we should not allow the callee side to move it.